ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -24,15 +27,207 @@ int TestListBuckets(const Aws::S3Crt::ClientConfiguration &config);
 int
 TestListBuckets(const Aws::S3Crt::ClientConfiguration &config)
 {
+    int ret = 0;
     S3Connection conn(config);
     std::vector<std::string> buckets;
-    if (!conn.ListBuckets(buckets))
-        return 1;
+    if (ret = conn.ListBuckets(buckets) != 0)
+        goto err;
 
     std::cout << "All buckets under my account:" << std::endl;
     for (const auto &bucket : buckets)
         std::cout << "  * " << bucket << std::endl;
-    return 0;
+
+err:
+    return ret;
+}
+
+/*
+ * TestListObjects --
+ *     Unit test for listing S3 objects under the first bucket in the associated AWS account. This
+ *     test assumes there are initially no objects with the prefix of "test_list_objects_" in the
+ *     bucket.
+ */
+int
+TestListObjects(const Aws::S3Crt::ClientConfiguration &config)
+{
+    int ret = 0;
+    S3Connection conn(config);
+
+    /* Temporary workaround to get a bucket to use. */
+    std::string firstBucket;
+    std::vector<std::string> buckets;
+
+    if (ret = conn.ListBuckets(buckets) != 0)
+        goto err;
+    if (!buckets.empty()) {
+        firstBucket = buckets.at(0);
+
+        std::vector<std::string> objects;
+        uint32_t countp;
+
+        /* No matching objects. */
+        if (ret = conn.ListObjects(firstBucket, "test_list_objects_", objects, countp) != 0)
+            goto err;
+        if (objects.size() != countp || countp != 0) {
+            ret = 1;
+            goto err;
+        }
+
+        /* Create file to prepare for test. */
+        if (!static_cast<bool>(std::ofstream("test_list_objects.txt").put('.'))) {
+            std::cerr << "Error creating file." << std::endl;
+            ret = 1;
+            goto err;
+        }
+
+        /* Put objects to prepare for test. */
+        for (int i = 0; i < 20; i++) {
```

Comment:
Can we define `20` in a variable? Seems like a magic number. I can see it on different lines below as well.